cmd/kubectl-gadget/deploy.go

```diff
@@ -472,6 +473,18 @@ func createAffinity(client *kubernetes.Clientset) (*v1.Affinity, error) {
 	return affinity, nil
 }
 
+func modifyPrometheusPort(ds *appsv1.DaemonSet, newPort string) error {
+	_, port, err := net.SplitHostPort(newPort)
+	if err != nil {
+		return fmt.Errorf("parsing port: %w", err)
+	}
+	if ds.Spec.Template.Annotations == nil {
+		ds.Spec.Template.Annotations = make(map[string]string)
+	}
+	ds.Spec.Template.Annotations["prometheus.io/port"] = port
+	return nil
+}
```

Comment:
nits:
1. newPort -> otelListenAddr?
2. Perhaps inlining this function is more clear.

Reply:
making updates, I think inline is better too